### PR TITLE
error handling middleware function added and used it in index.js

### PR DESCRIPTION
**Pull request by Dinesh-Khanal**
Hi, I have added errMiddleware.js file inside middleware folder and created notFound and errorHandler function. Imported these functions in index file and use them. I have tested making different error, it is working fine. Please check it and inform me if something goes wrong.

### CODE REVIEW

File: server/src/middleware/errMiddleware.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export function notFound(req, res, next) {
+  res.status(404);
+  const error = new Error(`Not Found - ${req.originalUrl}`);
+  next(error);
+}
+
+export function errorHandler(err, req, res, next) {
+  const statusCode = res.statusCode !== 200 ? res.statusCode : 500;
```
Comment (tsimian) [inline server/src/middleware/errMiddleware.js:8]: Could we not say `res.statusCode ? res.statusCode : 500`?

Comment (Dinesh-Khanal) [reply server/src/middleware/errMiddleware.js]: @tsimian Thank you for your suggestion. I have made change accordingly.

**Review by tsimian — APPROVED** [review]
Looks good to me @lugenx. Nicely done @Dinesh-Khanal!

**Review by lugenx — COMMENTED** [review]
thank you @Dinesh-Khanal it works perfectly, but my minor comments address just some formatting for readability and consistency. after they are done, it's good to go.

File: server/src/middleware/errMiddleware.js
Comment (lugenx) [inline server/src/middleware/errMiddleware.js:1]: also, let's change the filename to `errorHandler.js`

File: server/src/middleware/errMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export function notFound(req, res, next) {
+  res.status(404);
+  const error = new Error(`Not Found - ${req.originalUrl}`);
+  next(error);
+}
+
+export function errorHandler(err, req, res, next) {
```
[the remark this answers is not in the document]
Comment (Dinesh-Khanal) [reply server/src/middleware/errMiddleware.js]: Thank you @lugenx , I have changed according to your suggestion.

**Review by lugenx — APPROVED** [review]
thank you @Dinesh-Khanal

**Pull request merged**